lib/Jsrt/Jsrt.cpp

```diff
@@ -4293,11 +4293,28 @@ _ALWAYSINLINE JsErrorCode CompileRun(
 {
     PARAM_NOT_NULL(scriptVal);
     VALIDATE_JSREF(scriptVal);
+    PARAM_NOT_NULL(sourceUrl);
```

Comment:
This is a convenience matter we talked about offline. Previous Windows APIs use wchar_t* for sourceUrl and easy for caller to construct. Now it is harder. If we allow nullptr/JsInvalidRef and treat them as `""`, more convenient for many scenarios. Maybe discuss with limin.

---
In reply to: [87932689](https://github.com/Microsoft/ChakraCore/pull/1988#discussion_r87932689) [](ancestors = 87932689)